src/main/java/com/getindata/connectors/http/internal/table/lookup/CachingHttpTableLookupFunction.java

```diff
@@ -0,0 +1,88 @@
+package com.getindata.connectors.http.internal.table.lookup;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.table.connector.source.lookup.cache.LookupCache;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.functions.FunctionContext;
+import org.apache.flink.table.functions.LookupFunction;
+
+import com.getindata.connectors.http.internal.PollingClient;
+import com.getindata.connectors.http.internal.PollingClientFactory;
+import com.getindata.connectors.http.internal.utils.SerializationSchemaUtils;
+
+@Slf4j
+public class CachingHttpTableLookupFunction extends LookupFunction {
+    private final PollingClientFactory<RowData> pollingClientFactory;
+
+    private final DeserializationSchema<RowData> responseSchemaDecoder;
+
+    @VisibleForTesting
+    @Getter(AccessLevel.PACKAGE)
+    private final LookupRow lookupRow;
+
+    @VisibleForTesting
+    @Getter(AccessLevel.PACKAGE)
+    private final HttpLookupConfig options;
+
+    private transient AtomicInteger localHttpCallCounter;
+
+    private transient PollingClient<RowData> client;
+
+    private LookupCache cache;
```

Comment:
Cache is not used in this function

Reply:
yep, seems like you would like to use it in `lookup(RowData keyRow) ` method @davidradl 
Also, the `cache` field can be declared as final.